cloudinit/config/cc_apt_configure.py

```diff
@@ -6,225 +6,364 @@
+                        its entry in ``sources.list`` is not deleted; it
+                        is just commented out.""")
+                },
+                'primary': {
+                    **mirror_property,
+                    'description': dedent("""\
+                        The primary and security archive mirrors can
+                        be specified using the ``primary`` and
+                         ``security`` keys, respectively Both the
+                         ``primary`` and ``security`` keys take a list
+                        of configs, allowing mirrors to be specified
+                        on a per-architecture basis. Each config is a
+                        dictionary which must have an entry for
+                         ``arches``, specifying which architectures
+                        that config entry is for. The keyword
+                         ``default`` applies to any architecture not
+                        explicitly listed. The mirror url can be specified
+                        with the ``uri`` key, or a list of mirrors to
+                        check can be provided in order, with the first
+                        mirror that can be resolved being selected. This
+                        allows the same configuration to be used in
+                        different environment, with different hosts used
+                        for a local apt mirror. If no mirror is provided
+                        by ``uri`` or ``search``, ``search_dns`` may be
+                        used to search for dns names in the format
+                         ``<distro>-mirror`` in each of the following:
+
+                            fqdn of this host per cloud metadata,
+                            localdomain,
+                            domains listed in ``/etc/resolv.conf``.
+
+                        If there is a dns entry for ``<distro>-mirror``,
+                        then it is assumed that there is a distro mirror
+                        at ``http://<distro>-mirror.<domain>/<distro>``.
+                        If the ``primary`` key is defined, but not the
+                         ``security`` key, then then configuration for
+                         ``primary`` is also used for ``security``.
+                        If ``search_dns`` is used for the ``security``
+                         key, the search pattern will be
+                         ``<distro>-security-mirror``.
+
+                        If no mirrors are specified, or all lookups fail,
+                        then default mirrors defined in the datasource
+                        are used. If none are present in the datasource
+                        either the following defaults are used:
+
+                        ``primary`` => ``http://archive.ubuntu.com/ubuntu``,
+                         ``security`` =>
+                         ``http://security.ubuntu.com/ubuntu``
```

Comment:
White space damage to sort separately in cloudinit/config/schema.py

And make this a list of bullets
```suggestion
                        - ``primary`` => ``http://archive.ubuntu.com/ubuntu``,
                         - ``security`` => ``http://security.ubuntu.com/ubuntu``
```

Reply:
+  1